backend/pkg/k8cache/cacheStore.go

```diff
@@ -134,3 +135,19 @@ func GenerateKey(url *url.URL, contextID string) (string, error) {
 
 	return raw, nil
 }
+
+// UnmarshalCachedData deserialize a JSON string received from cache
+// back into a CachedResponseData struct. This function is used to reconstructing
+// the full HTTP response (status, headers, body) when serving the k8s to the client.
+// this is the essential part as it gives the clarity about the incoming k8s requests.
+func UnmarshalCacheData(cacheResource string,
```

Comment:
Please remove UnmarshalCacheData.

It is only used in the `LoadFromCache` function, and it doesn't add anything extra to justify a function.

```go
var cachedData CachedResponseData
cachedData, err := UnmarshalCacheData(k8Resource)
if err != nil {
	return false, err
}
```

If you just use json.Unmarshal it's about the same:
```go
var cachedData CachedResponseData
if err := json.Unmarshal([]byte(k8Resource), &cachedData); err != nil {
  return false, err
}
```